src/template.cpp

```diff
@@ -0,0 +1,146 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// Positive substring: return a substring of `str``going from `start` to `end`;
+// if `end` is earlier than `start`, return empty string.
+std::string pos_substr(std::string str, int start, int end) {
+  if (end < start)
+    return "";
+  else
+    return str.substr(start, end - start + 1);
+}
+
+// Break template text into character vector. The first element element of the
+// resulting vector is HTML, the next is R code, and they continue alternating.
+// [[Rcpp::export]]
+CharacterVector template_dfa(CharacterVector x) {
```

Comment:
You can just declare the parameter as std::string if that's what you want to ultimately handle (the Rcpp::as will be done behind the scenes)